src/elm/arithmetic.ts

```diff
@@ -391,9 +392,13 @@ export class MinValue extends Expression {
 
   valueType: keyof typeof MinValue.MIN_VALUES;
 
-  constructor(json: any) {
+  constructor(json: ELM.MinValue) {
     super(json);
-    this.valueType = json.valueType;
+    if (json.valueType in MinValue.MIN_VALUES) {
+      this.valueType = json.valueType as keyof typeof MinValue.MIN_VALUES;
```

Comment:
Small (opinionated) comment. Now that we use `keyof typeof MinValue.MIN_VALUES` in more than one place, maybe we can bring it out to its own `type` outside of the class? E.g.:

``` TypeScript
type MinValueType = keyof typeof MinValue.MIN_VALUES;
```

Then we can just re-use this type where applicable. Same applies to max values below

Reply:
It also does make sense to pull MIN_VALUE out, and maybe even use an ENUM instead of a dict. But it was a little weird to even have to use `as keyof typeof MinValue.MIN_VALUES`... I thought Typescript would be able to infer that since `json.valueType in MinValue.MIN_VALUES` must be true